app/models/user.rb

```diff
@@ -1122,6 +1116,18 @@ def mature_staged?
     from_staged? && self.created_at && self.created_at < 1.day.ago
   end
 
+  def available_titles
+    group_titles_query = groups
+    group_titles_query = group_titles_query.order("groups.id = #{primary_group_id} DESC") if primary_group_id
+    group_titles_query = group_titles_query.order("groups.primary_group DESC")
+
+    group_titles = group_titles_query.pluck(:title).reject(&:blank?)
```

Comment:
hmm perhaps we can check for a blank title when fetching the recrod?

Reply:
wanted to do it in the query, but there isn't an index on `title`.

Reply:
Hmm I think this needs to be optimized further, the only place we're using this method calls `.first` on it. That means that we actually don't need to fetch all the titles out of the DB.

Reply:
For sure, I will change the behavior and move this to `GroupUser` then!